R/ts_plot.R

```diff
@@ -1,19 +1,38 @@
+library(ggplot2)
+library(forecast)
 #' Time series decomposition and plot
 #'
 #' Convert csv file into a time series object, decompose it into in trend, seasonality/cyclicity
 #' and noise/remainder/error and plot the raw data and the decomposed components.
 #'
 #' @param data str: the path of the csv file
-#' @param type str: "regular" or "irregular"
-#'            "regular": time series with regular time intervals
-#'            "irregular": time series with irregular time intervals
 #' @param col str: the column to be analyzed
 #' @param frequency int: the desired time interval
 #'
-#' @return a figure with 4 subplots
+#' @return ggplot Object
 #' @export
 #' @examples
-#' ts_plot("https://raw.githubusercontent.com/plotly/datasets/master/timeseries.csv", A, irregular, 3)
-ts_plot <- function(data, col, type, frequency){
-  print("hello")
+#' ts_plot("~/Desktop/ts4_jjshares.csv", "earnings", 4)
+ts_plot <- function(data, col, frequency){
+  df <- readr::read_csv(data)
+  ts <- ts(df[[col]], frequency = frequency)
+
+  if (time(ts)[2] - time(ts)[1] != time(ts)[3] - time(ts)[2]){
+    stop("This time series is irregular.")
+  }
+
+  if (time(ts)[2] - time(ts)[1] != 1/frequency){
+    stop("The frequency is wrong.")
```

Comment:
You could potentially combine these into the same if/else statement, although it is fine the way it is as well. Maybe add more descriptive error messages to help guide the user in debugging.

Reply:
OK, I have added more descriptive error messages to it.